### PR TITLE
Added Protein Data Bank citation reader

### PR DESCRIPTION
**Pull request by joshuakto**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Added feature to read primary citation of Protein Data Bank entries.

## Type of Change

Please delete options that are not relevant.

- [X] New Loader/Tool

# How Has This Been Tested?

 - Ran "poetry run pytest tests" to check that the added loader does not introduce new errors/warnings.
 - Added tests/test_pdb.py to test that the dataloader behaves as expected and return the right document object.

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [X] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [X] I have added a library.json file if a new loader/tool was added
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [N/A] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: tests/test_pdb/test_pdb.py
Comment (EmanuelCampos) [inline tests/test_pdb/test_pdb.py:1]: can we mock this HTTP request? then we avoid the CI breaks if the service goes down

File: llama_hub/pdb/__init__.py
Comment (EmanuelCampos) [inline llama_hub/pdb/__init__.py:1]: """Init file."""
```py
from llama_hub.pdb.base import (
    PdbAbstractReader
)

__all__ = ["PdbAbstractReader"]
```

**Review by EmanuelCampos — COMMENTED** [review]
did two comments, and the tests are broken, can you fix that? thanks for the PR

**Comment by joshuakto** [comment]
Thanks @EmanuelCampos! Updated the code base on the two comments, please feel free to suggest additional improvements.

**Review by EmanuelCampos — APPROVED** [review]

**Review by anoopshrma — APPROVED** [review]

**Pull request merged**